test/lish/cloud_storage_upload_provides_an_error_test.dart

```diff
@@ -22,14 +22,30 @@ void main() {
     globalServer.expect('POST', '/upload', (request) {
       return request.read().drain().then((_) {
         return shelf.Response.notFound(
-          '<Error><Message>Your request sucked.</Message></Error>',
+          // Actual example of an error code we get from GCS
+          "<?xml version='1.0' encoding='UTF-8'?><Error><Code>EntityTooLarge</Code><Message>Your proposed upload is larger than the maximum object size specified in your Policy Document.</Message><Details>Content-length exceeds upper bound on range</Details></Error>",
           headers: {'content-type': 'application/xml'},
         );
       });
     });
 
-    // TODO(nweiz): This should use the server's error message once the client
-    // can parse the XML.
+    expect(
+      pub.stderr,
+      emits(
+        'Server error code: EntityTooLarge',
+      ),
+    );
+    expect(
+      pub.stderr,
+      emits(
+        'Server message: Your proposed upload is larger than the maximum object size specified in your Policy Document.',
+      ),
+    );
+    expect(
+      pub.stderr,
+      emits('Server details: Content-length exceeds upper bound on range'),
+    );
```

Comment:
What does this look like in a terminal?

Are you sure we don't want to special case this?

Reply:
> What does this look like in a terminal?

<img width="1019" alt="image" src="https://github.com/dart-lang/pub/assets/8613953/17dfb3fe-7f23-4073-b953-9340e5d92b51">

> Are you sure we don't want to special case this?

Such that we recognize this specific message and give our own reply? That might work as well...

Reply:
I think special-casing for specific messages would be neat - but also risks swallowing specific details from the error message.

I'll add a TODO for now.